app/tests/test_activity_api.py

```diff
@@ -0,0 +1,51 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from rest_framework.test import APIClient
+from rest_framework import status
+
+from core.models import Activity
+from api.serializers import ActivitySerializer
+
+from tests import create_new_user, create_new_project, create_new_activity
+
+
+class PrivateActivityTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = create_new_user("Seba")
+        self.client.force_authenticate(self.user)
+        self.project = create_new_project(user=self.user, title="Learn Python")
+        self.activity = create_new_activity(project=self.project)
+        self.general_url = reverse('api:activity-list')
```

Comment:
I would not have properties like `self.general_url` or `self.detail_url` or even `self.activity` but rather call them directly per test case, it does not make sense to ie. create project or create details url variable if you are not planning to use it.

Example of this is `test_starting_activity` test case. You do not use `details_url` there - so whats the point of creating it inside setUp?

Inside setUp you should create and assign to self ONLY structures you plan on reusing in ALL your test cases.

Reply:
OK, changed